pytorch_lightning/profiler/profilers.py

```diff
@@ -466,15 +466,22 @@ def _create_profiler(self, action_name, profiler, enter=True):
         profiler_args = {k: v for k, v in vars(self).items() if k in init_args}
         pr = profiler(**profiler_args)
         if enter:
-            pr = pr.__enter__()
+            out_pr = pr.__enter__()
+            if out_pr is not None:
+                pr = out_pr
         self.profiler = pr
+        return pr
 
     def _stop(self, action_name: str) -> None:
         if self.profiler is None:
             return
 
         self.profiler.__exit__(exc_type=None, exc_val=None, exc_tb=None)
 
+        if isinstance(self.profiler, torch.autograd.profiler.emit_nvtx):
+            self._profiler.__exit__(None, None, None)
```

Comment:
it is a bit confusing, what is diff between `self.profiler` and `self._profiler`

Reply:
Renamed to parent_profiler.